TowerNinja/Assets/Scripts/Arrow.cs

```diff
@@ -27,6 +29,19 @@ void Update()
     {
         if (transform.position.y < -4.5) Die();
     }
+    
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.CompareTag("enemy"))
+        {
+            Debug.LogError("Enemy Collision");
+            Debug.Log(collision.gameObject);
+            var parent = collision.gameObject.transform.parent;
+            parent.GetComponent<enemyDmgHitbox>()?.DamageEnemy(damage);
+            FindObjectOfType<ResourceManager>().UpdateProjectileDamage("arrow",damage);
```

Comment:
I also mentioned this in #58 . I did some research and it seems like FindObjectOfType() has very slow performance because it does hierarchical search through the entire scene, and is discouraged in general cases. A new arrow is spawned every second so we probably want to change this